Python/compile.c

```diff
@@ -7375,10 +7415,14 @@ _PyCompile_Assemble(_PyCompile_CodeUnitMetadata *umd, PyObject *filename,
         goto error;
     }
 
-    PyObject *consts = umd->u_consts;
+    PyObject *consts = consts_dict_keys_inorder(umd->u_consts);
+    if (consts == NULL) {
+        goto error;
+    }
```

Comment:
Just to make sure I'm understanding what's happening here:

1. `u_metadata.u_consts` is normally a dict in the compiler.
2. But here we passed it to `_PyAssemble_MakeCodeObject`, which expects to get a list, not a dict.
3. This only worked before because `_PyCompile_Assemble` is only used from unit tests, where we were passing in a manually-constructed metadata object where we (inconsistent with the compiler) constructed `u_consts` as a list.
4. Now that we're actually passing through the metadata from the compiler to `_PyCompile_Assemble` in some tests, the inconsistency was revealed, so we're fixing this bug (and updating the test where we manually construct metadata for this API to use a dict instead.)

Reply:
That’s right.